### PR TITLE
Fix loses index during document updates

### PR DESCRIPTION
**Pull request by mad**
This PR prepare fix only for #764. Partially based on this PR #929

After merge I will prepare PR for #670
 
Signed-off-by: Pavel Ershov <owner.mad.epa@gmail.com>

-----

Thank you for contributing to JanusGraph!

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there an issue associated with this PR? Is it referenced in the commit message?
- [ ] Does your PR body contain #xyz where xyz is the issue number you are trying to resolve?
- [ ] Has your PR been rebased against the latest commit within the target branch (typically `master`)?
- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] Have you written and/or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](https://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE.txt file, including the main LICENSE.txt file in the root of this repository?
- [ ] If applicable, have you updated the NOTICE.txt file, including the main NOTICE.txt file found in the root of this repository?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?
- [ ] If this PR is a documentation-only change, have you added a `[skip ci]`
  tag to the first line of your commit message to avoid spending CPU cycles in
  Travis CI when no code, tests, or build configuration are modified?

### Note:
Please ensure that once the PR is submitted, you check Travis CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: janusgraph-test/src/main/java/org/janusgraph/diskstorage/indexing/IndexProviderTest.java [hunk 1]
```diff
@@ -1057,6 +1057,63 @@ public void testScroll() throws BackendException {
...
+        // check every indexed type is still in the index
+        assertEquals(1, tx.queryStream(new IndexQuery(store, PredicateCondition.of(DATE, Cmp.EQUAL, Instant.ofEpochSecond(1)))).count());
+        assertEquals(1, tx.queryStream(new IndexQuery(store, PredicateCondition.of(LOCATION, Geo.WITHIN, Geoshape.box(latitude-1L, longitude-1, latitude+1, longitude+1)))).count());
+        assertEquals(1, tx.queryStream(new IndexQuery(store, PredicateCondition.of(WEIGHT, Cmp.EQUAL, 1.5d))).count());
+        assertEquals(1, tx.queryStream(new IndexQuery(store, PredicateCondition.of(TIME, Cmp.EQUAL, 1L))).count());
+        assertEquals(1, tx.queryStream(new IndexQuery(store, PredicateCondition.of(STRING, Cmp.EQUAL, "network"))).count());
+        assertEquals(1, tx.queryStream(new IndexQuery(store, PredicateCondition.of(KEYWORD, Text.CONTAINS, "network"))).count());
+        assertEquals(1, tx.queryStream(new IndexQuery(store, PredicateCondition.of(ANALYZED, Text.CONTAINS, "network"))).count());
+        assertEquals(1, tx.queryStream(new IndexQuery(store, PredicateCondition.of(NAME, Cmp.EQUAL, "network"))).count());
+
```
Comment (porunov) [inline janusgraph-test/src/main/java/org/janusgraph/diskstorage/indexing/IndexProviderTest.java]: I see that you partially copied the test from #929 but you removed the next lines from the test:
```
assertEquals(1, tx.queryStream(new IndexQuery(store, PredicateCondition.of(BOOLEAN, Cmp.EQUAL, true))).count());
assertEquals(1, tx.queryStream(new IndexQuery(store, PredicateCondition.of(TRUEDATE, Cmp.EQUAL, date))).count());
```

Is there a reason why you didn't include these assertions?

Comment (mad) [reply janusgraph-test/src/main/java/org/janusgraph/diskstorage/indexing/IndexProviderTest.java]: TRUEDATE and BOOLEAN indexed is same as DATE, NUMERIC for test simplicity I removed thats fields

Comment (porunov) [reply janusgraph-test/src/main/java/org/janusgraph/diskstorage/indexing/IndexProviderTest.java]: Than `Date date = new Date(1L);` isn't needed as suggested by Codacy

Comment (mad) [reply janusgraph-test/src/main/java/org/janusgraph/diskstorage/indexing/IndexProviderTest.java]: Done

**Comment by porunov** [comment]
@mad your verification is needed here #1683

**Pull request merged**